src/viam/sdk/resource/reconfigurable.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/resource/resource.hpp>
+
+namespace viam {
+namespace sdk {
+
+class Reconfigurable {
+   public:
+    virtual ~Reconfigurable();
+
+    /// @brief Reconfigures a resource.
+    /// @param deps Dependencies of the resource.
+    /// @param cfg The resource's config.
+    virtual void reconfigure(const Dependencies& deps, const ResourceConfig& cfg) = 0;
```

Comment:
I opted to change the signature of `reconfigure` to take its args by `const&`. I think this especially makes sense for the cfg; I could theoretically imagine someone wanting non-`const` access to `deps` but I think passing `deps` by value would be a bit strange, and I suspect that `const&` dependencies will be correct and manageable. In the worst case, we can add an override of `reconfigure` that isn't `const&`. Happy to revisit if others have thoughts.

Reply:
I think it is fine.